### ISSUE TITLE
Feature request: better way to select what top-level view to render 

### ISSUE THREAD

**Issue by lessless** [phoenixframework/phoenix_live_view#2106]
Hi, 

We have two different user types - a Clinician and a Patient, and both have a dashboard under the same `/dashboard` route. 
As those are two completely different user roles, their dashboards have other UI controls and present other data sets. So just rendering separate templates won't cut it.

Differentiating between those two use cases is not a problem in a classic controller - just pattern match on the user's role field in a controller action and then do your thing. But it seems to be a bit harder with LV.

Right now, we are using `live_render` and having a problem with catching `phx:page-loading-stop`

```elixir
defmodule DashboardController do

   def show(%{assigns: %{current_user: %{role: :patient}}} = conn, params) do
     live_render(
       conn,
       PatientDashboardLive
     )
   end

   def show(%{assigns: %{current_user: %{role: :clinician}}} = conn, params) do
     live_render(
       conn,
       ClinicianDashboardLive
     )
   end
 end
```

Which has its own trade-offs, as @chrismccord  wrote in Slack

> you forgo a lot with live_render in the controller
> you basically almost never want to do that
> live navigation is not supported in such cases, so the page lifecycle is not in our control in that case

We can convert them to LiveComponents and decide which one to pick in `render` callback, but that has drawbacks of sharing the parent process, which makes `handle_info`/PubSub/message passing quite noisy. 

Also, I've heard suggestions to split them by different URLs, but that seems to bring about quite a bit of hopefully unnecessary technical burden. The chances of real users having two accounts under two different roles in our use case is close to zero (it's quite a niche system). Therefore, it will violate the principle of the least surprise for everyone who doesn't have context in their heads when reading the code. 
Also, we will be looking at additional security concerns with authorisation. 

The bottom line is there is no "ideal" solution when it comes down to choosing which top-level LiveView we should render at the given route. And since LiveView combines presentation and logic, this issue is of a different magnitude from classical controllers. 

I understand that LiveView is another model with its own trade-offs, and this might not be the immediate priority, yet I hope we will have better options.

**Comment by josevalim** [comment]
The better option is indeed separate routes. I don't think it should cause additional concerns because I assume your application already has other routes that are specific to clinician or patient and you already have to protect them. You should also be using `live_session` to make sure a role cannot migrate from one live session to another.

The only drawback really of using separate routes is that now you have to redirect to two different places, but that can be easily encapsulated in a function that receives the user type and chooses accordingly.

You have the trade-offs from both cases, so it is up to you to choose the one you prefer.

**Comment by lessless** [comment]
Actually, we have an absolute LiveView routing parity for now, and that would be the first case when we will need to introduce `live_session`.
